src/Lifecycle.js

```diff
@@ -462,10 +470,28 @@ async function _doSetLoggedIn(credentials, clearStorage) {
 
     Analytics.setLoggedIn(credentials.guest, credentials.homeserverUrl);
 
+    MatrixClientPeg.replaceUsingCreds(credentials);
+    const client = MatrixClientPeg.get();
+
+    if (credentials.freshLogin) {
```

Comment:
AFAIK, we have not yet decided that we actually want this enable for all users by default... Even if we do eventually want that, I think we'll want more control over the rollout process. Please add a [feature flag](https://github.com/vector-im/element-web/blob/develop/docs/feature-flags.md) to give us some control here. Given that it happens at login time, it's likely not very useful as UI setting post-login, but it still offers config-level control over whether it's enabled for a deployment.

Reply:
I think it should be safe to dehydrate a device if one exists, since the user has already decided to create a dehydrated device.  But I've added a flag for where it creates a dehydrated device when a device didn't already exist.

Reply:
Thanks for adding the feature flag. I think it makes sense to guard all of the related paths (including this one), since they are part of the feature being rolled out.

For example, if `rehydrateDevice` is supported by the server, but happens to have a bug where it hangs, we would want to be able to disable this in Element until it can be fixed.

Reply:
fair enough